src/script_interface/reaction_methods/ReactionAlgorithm.hpp

```diff
@@ -56,19 +56,38 @@ class ReactionAlgorithm : public AutoParameters<ReactionAlgorithm> {
   virtual std::shared_ptr<::ReactionMethods::ReactionAlgorithm> RE() = 0;
 
   ReactionAlgorithm() {
-    add_parameters({
-        {"reactions", AutoParameter::read_only,
-         [this]() {
-           std::vector<Variant> out;
-           for (auto const &e : m_reactions) {
-             out.emplace_back(e);
-           }
-           return out;
-         }},
-        {"kT", AutoParameter::read_only, [this]() { return RE()->get_kT(); }},
-        {"exclusion_radius", AutoParameter::read_only,
-         [this]() { return RE()->get_exclusion_radius(); }},
-    });
+    add_parameters(
+        {{"reactions", AutoParameter::read_only,
+          [this]() {
+            std::vector<Variant> out;
+            for (auto const &e : m_reactions) {
+              out.emplace_back(e);
+            }
+            return out;
+          }},
+         {"kT", AutoParameter::read_only, [this]() { return RE()->get_kT(); }},
+         {"exclusion_range", AutoParameter::read_only,
+          [this]() { return RE()->get_exclusion_range(); }},
+         {"exclusion_radius_per_type",
+          [this](Variant const &v) {
+            std::unordered_map<int, double> exclusion_radius_per_type;
+
+            for (auto const &item :
+                 get_value<std::unordered_map<int, Variant>>(v)) {
+              exclusion_radius_per_type[item.first] = get_value<double>(
+                  get_value<std::unordered_map<int, Variant>>(v).at(
+                      item.first));
+            }
```

Comment:
How about the following?
```c++
  get_map<int, double>(get_value_or<std::unordered_map<int, Variant>>(
      v, "exclusion_radius_per_type", {}))
```

Reply:
Maybe I don't quite get how it should work but this line
`{"exclusion_radius_per_type",
          [this](Variant const &v) {
            RE()->exclusion_radius_per_type =   get_map<int, double>(get_value_or<std::unordered_map<int, Variant>>(
      v, "exclusion_radius_per_type", {}));
          }`
gives me the following error:
     In file included from /home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/ConstantpHEnsemble.hpp:23,
                 from /home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/initialize.cpp:24:
/home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/ReactionAlgorithm.hpp: In lambda function:
/home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/ReactionAlgorithm.hpp:76:41: error: no matching function for call to ‘get_value_or<std::unordered_map<int, boost::variant<boost::detail::variant::recursive_flag<ScriptInterface::None>, bool, int, long unsigned int, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<int, std::allocator<int> >, std::vector<double, std::allocator<double> >, std::shared_ptr<ScriptInterface::ObjectHandle>, std::vector<boost::recursive_variant_, std::allocator<boost::recursive_variant_> >, Utils::Vector<double, 2>, Utils::Vector<double, 3>, Utils::Vector<double, 4>, std::unordered_map<int, boost::recursive_variant_, std::hash<int>, std::equal_to<int>, std::allocator<std::pair<const int, boost::recursive_variant_> > > >, std::hash<int>, std::equal_to<int>, std::allocator<std::pair<const int, boost::variant<boost::detail::variant::recursive_flag<ScriptInterface::None>, bool, int, long unsigned int, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<int, std::allocator<int> >, std::vector<double, std::allocator<double> >, std::shared_ptr<ScriptInterface::ObjectHandle>, std::vector<boost::recursive_variant_, std::allocator<boost::recursive_variant_> >, Utils::Vector<double, 2>, Utils::Vector<double, 3>, Utils::Vector<double, 4>, std::unordered_map<int, boost::recursive_variant_, std::hash<int>, std::equal_to<int>, std::allocator<std::pair<const int, boost::recursive_variant_> > > > > > > >(const Variant&, const char [26], <brace-enclosed initializer list>)’
       v, "exclusion_radius_per_type", {}));
                                         ^
In file included from /home/pablo/Desktop/espresso_pablo/src/script_interface/auto_parameters/AutoParameter.hpp:23,
                 from /home/pablo/Desktop/espresso_pablo/src/script_interface/auto_parameters/AutoParameters.hpp:24,
                 from /home/pablo/Desktop/espresso_pablo/src/script_interface/ScriptInterface.hpp:30,
                 from /home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/SingleReaction.hpp:24,
                 from /home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/initialize.cpp:22:
/home/pablo/Desktop/espresso_pablo/src/script_interface/get_value.hpp:301:3: note: candidate: ‘template<class T> T ScriptInterface::get_value_or(const VariantMap&, const string&, const T&)’
 T get_value_or(VariantMap const &vals, std::string const &name,
   ^~~~~~~~~~~~
/home/pablo/Desktop/espresso_pablo/src/script_interface/get_value.hpp:301:3: note:   template argument deduction/substitution failed:
In file included from /home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/ConstantpHEnsemble.hpp:23,
                 from /home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/initialize.cpp:24:
/home/pablo/Desktop/espresso_pablo/src/script_interface/reaction_methods/ReactionAlgorithm.hpp:76:7: note:   cannot convert ‘v’ (type ‘const Variant’ {aka ‘const boost::variant<boost::detail::variant::recursive_flag<ScriptInterface::None>, bool, int, long unsigned int, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<int, std::allocator<int> >, std::vector<double, std::allocator<double> >, std::shared_ptr<ScriptInterface::ObjectHandle>, std::vector<boost::recursive_variant_, std::allocator<boost::recursive_variant_> >, Utils::Vector<double, 2>, Utils::Vector<double, 3>, Utils::Vector<double, 4>, std::unordered_map<int, boost::recursive_variant_, std::hash<int>, std::equal_to<int>, std::allocator<std::pair<const int, boost::recursive_variant_> > > >’}) to type ‘const VariantMap&’ {aka ‘const std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<boost::detail::variant::recursive_flag<ScriptInterface::None>, bool, int, long unsigned int, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<int, std::allocator<int> >, std::vector<double, std::allocator<double> >, std::shared_ptr<ScriptInterface::ObjectHandle>, std::vector<boost::recursive_variant_, std::allocator<boost::recursive_variant_> >, Utils::Vector<double, 2>, Utils::Vector<double, 3>, Utils::Vector<double, 4>, std::unordered_map<int, boost::recursive_variant_, std::hash<int>, std::equal_to<int>, std::allocator<std::pair<const int, boost::recursive_variant_> > > > >&’}
       v, "exclusion_radius_per_type", {}));
       ^
make[2]: *** [src/script_interface/CMakeFiles/ScriptInterface.dir/build.make:323: src/script_interface/CMakeFiles/ScriptInterface.dir/reaction_methods/initialize.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:6649: src/script_interface/CMakeFiles/ScriptInterface.dir/all] Error 2
make: *** [Makefile:141: all] Error 2